src/main/java/org/mule/extension/sftp/internal/operation/SftpCommand.java

```diff
@@ -287,11 +287,6 @@ protected String getCurrentWorkingDirectory() {
    * {@inheritDoc}
    */
   protected URI getBasePath(FileSystem fileSystem) {
-    String basePath = fileSystem.getBasePath();
-    if (isEmpty(basePath)) {
-      basePath = ((SftpFileSystemConnection) fileSystem).getClient().getHome();
-    }
-    return createUri(basePath);
```

Comment:
Isn't this breaking backwards compatibility?
If the problem s that the pwd commands hangs... isn't it better to set a timeout for it (I know is not possible due the Mina API, but you can wrap and timeout from this side).
CC: @elcoya // @acurci-at-mulesoft